perun-core/src/test/java/cz/metacentrum/perun/core/entry/ResourcesManagerEntryIntegrationTest.java

```diff
@@ -494,39 +495,117 @@ public void assginGroupToResource() throws Exception {
+
+	}
+
+	@Test
+	public void assignGroupToResourceAsInactive() throws Exception {
+		System.out.println(CLASS_NAME + "assignGroupToResourceAsInactive");
+
+		vo = setUpVo();
+		facility = setUpFacility();
+		resource = setUpResource();
+		assertNotNull("unable to create resource", resource);
+		member = setUpMember(vo);
+		group = setUpGroup(vo, member);
+		resourcesManager.assignGroupToResource(sess, group, resource, false, true, false);
+
+		AssignedGroup expectedGroup = new AssignedGroup(new EnrichedGroup(group, null), GroupResourceStatus.INACTIVE, null, null, false);
+
+		List<AssignedGroup> assignedGroups = resourcesManager.getGroupAssignments(sess, resource, null);
+		assertEquals("one group should be assigned to our Resource", 1, assignedGroups.size());
+		assertTrue("our group should be assigned to resource Expected: " + group.getName() + ", Actual: " + assignedGroups.get(0).getEnrichedGroup().getGroup().getName(), assignedGroups.contains(expectedGroup));
+		assertEquals("our group should be assigned to resource as inactive", assignedGroups.get(0).getStatus(), GroupResourceStatus.INACTIVE);
+	}
+
+	@Test
+	public void assignGroupToResourceWithSubgroup() throws Exception {
+		System.out.println(CLASS_NAME + "assignGroupToResourceWithSubgroup");
+
+		vo = setUpVo();
+		facility = setUpFacility();
+		resource = setUpResource();
+		assertNotNull("unable to create resource", resource);
+		member = setUpMember(vo);
+
+		group = setUpGroup(vo, member);
+		subGroup = setUpSubGroup(group);
+		AssignedGroup expectedGroup = new AssignedGroup(new EnrichedGroup(group, null), GroupResourceStatus.ACTIVE, null, null, true);
+		AssignedGroup expectedGroupProcessing = new AssignedGroup(new EnrichedGroup(group, null), GroupResourceStatus.PROCESSING, null, null, true);
+		AssignedGroup expectedSubGroup = new AssignedGroup(new EnrichedGroup(subGroup, null), GroupResourceStatus.ACTIVE, expectedGroup.getEnrichedGroup().getGroup().getId(), null, true);
+		AssignedGroup expectedSubGroupProcessing = new AssignedGroup(new EnrichedGroup(subGroup, null), GroupResourceStatus.PROCESSING, expectedGroup.getEnrichedGroup().getGroup().getId(), null, true);
+
+		resourcesManager.assignGroupToResource(sess, group, resource, false, false, true);
+
+		List<AssignedGroup> assignedGroups = resourcesManager.getGroupAssignments(sess, resource, null);
+		assertEquals("two groups (group with subgroup) should be assigned to our Resource", 2, assignedGroups.size());
 
+		assertTrue("Our group should be assigned to resource.", assignedGroups.contains(expectedGroup) || assignedGroups.contains(expectedGroupProcessing));
+		assertTrue("Our subgroup should be assigned to resource.", assignedGroups.contains(expectedSubGroup) || assignedGroups.contains(expectedSubGroupProcessing));
+
+		for (AssignedGroup group : assignedGroups) {
+			assertTrue("All groups should have 'autoAssignSubgroups' flag set to true", group.isAutoAssignSubgroups());
```

Comment:
We should probably check this only for the main group. This flag is irrelevant for automatic subgroup assignments.